sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/UserIdContext.java

```diff
@@ -19,6 +19,33 @@
     @VisibleForTesting
     public static final int USER_ID_APP_CENTER_MAX_LENGTH = 256;
 
+    /**
+     * Unique instance.
+     */
+    private static UserIdContext sInstance;
```

Comment:
`UserIdContext` only has one string property which is `userId`. I don't think `getInstance()` is necessary unless we will have other properties here in the future.
`private static String sUserId;`??

Reply:
@jaelim-ms that's how we originally implemented it in #890 - we wanted to align with the Apple implementation.  Do you think we should close this instead of merging it?

Reply:
@jaelim-ms there could be history later for breakpad and aligning with Apple and best practices for shared objects.

Reply:
@guperrot Makes sense.